spec/models/setting_spec.rb

```diff
@@ -3,8 +3,35 @@
 RSpec.describe Setting, type: :model do
   context '.get' do
     it 'returns the specified setting' do
-      create :setting, name: 'foo', value: 'bar'
-      expect( Setting.get( 'foo' ) ).to eq 'bar'
+      setting = create :setting, name: 'foo'
+      create :setting_value, setting_id: setting.id, value: 'bar'
+
+      expect( Setting.get( :foo ) ).to eq 'bar'
+    end
+
+    it 'returns the specified admin setting if user is an admin' do
+      setting = create :setting, name: 'foo', level: 'admin'
+      create :setting_value, setting_id: setting.id, value: 'bar'
+      user = create :admin_user
+
+      expect( Setting.get( :foo, user ) ).to eq 'bar'
     end
   end
+
+  it 'fails validation when updating a locked setting' do
+    setting = create :setting, name: 'foo', level: 'site', locked: true
+
+    _null = setting.update( level: 'user' )
```

Comment:
Consider adding an expectation on this line too, testing `#update` returns false when `setting` is invalid:

```diff
-_null = setting.update( level: 'user' )
+expect(setting.update( level: 'user' )).to eq false
```